api/proquest/client.py

```diff
@@ -459,6 +481,39 @@ def _download_feed_page(self, configuration, page, hits_per_page):
 
         return response_json[self.RESPONSE_OPDS_FEED_FIELD]
 
+    @staticmethod
+    def _is_feed_page_empty_or_incorrect(feed: Dict) -> bool:
+        """Determine whether the feed page is correctly structured and contains any publications.
+
+        ProQuest used to throw a 404 or 500 error when we tried to request a non-existent page,
+        but this behavior recently had changed. Instead of throwing an error, they
+        return empty pages, making CM infinitely loop through them.
+
+        NOTE: This method doesn't use a proper OPDS 2.x parser because it'd slow down downloading a big feed.
+        Instead, it uses a number of heuristics we got working with the ProQuest feed.
+        If the structure of the feed changes, this method will stop working and the download process will fail.
```

Comment:
Within each page of the feed (most importantly, the first page), we get `.opdsFeed.metadata.numberOfItems`. Could we use that value, along with the current number of items per page to calculate the correct number of pages to fetch?

Reply:
I wouldn't trust the ProQuest pagination mechanism, it doesn't work correctly.

Reply:
Jonathan Wyrembelski from ProQuest proposed the same but I'd like to make sure that feed's metadata is correct and do it in a separate PR.

Reply:
I won't hold up this PR for this.